pkg/testrunner/list.go

```diff
@@ -108,7 +108,9 @@ func (rl RunList) Run(log logr.Logger, config *Config, testrunNamePrefix string,
 			}
 			*rl[trI] = *newRun
 			attempt++
-		})
+			executor.AddItem(f)
+		}
+		executor.AddItem(f)
```

Comment:
Can you explain why you need to add f twice? Looks confusing

Reply:
I add `f` once for the initial run and then for the retries inside the execution